### DISCUSSION TITLE
Print nesteded derivative functions

### DISCUSSION THREAD

**Discussion by MaxSagebaum** [vgvassilev/clad#1155] [General]
I have an example with a nested function call:
```
#include "clad/Differentiator/Differentiator.h"
#include <iostream>

double inner(double x) {
  return x * x;
}

void f(double const* x, double* y) {
  for(int i = 0; i < 4; i += 4) {
    y[i] = inner(x[i]);
  }
}


int main() {

  auto f_dx = clad::gradient(f);        // does not dectet y as in input
  f_dx.dump();
}
```

The output of the program is:
```
The code is: 
void f_grad(const double *x, double *y, double *_d_x, double *_d_y) {
    int _d_i = 0;
    int i = 0;
    clad::tape<int> _t1 = {};
    clad::tape<double> _t2 = {};
    unsigned long _t0 = 0UL;
    for (i = 0; ; clad::push(_t1, i) , (i += 4)) {
        {
            if (!(i < 4))
                break;
        }
        _t0++;
        clad::push(_t2, y[i]);
        y[i] = inner(x[i]);
    }
    for (;; _t0--) {
        {
            if (!_t0)
                break;
        }
        {
            i = clad::pop(_t1);
            int _r_d0 = _d_i;
        }
        {
            y[i] = clad::pop(_t2);
            double _r_d1 = _d_y[i];
            _d_y[i] = 0.;
            double _r0 = 0.;
            inner_pullback(x[i], _r_d1, &_r0);
            _d_x[i] += _r0;
        }
    }
}
```

The function `inner_pullback` is missing in the output. Is there a way to print it? I could not find anything in the documentation on https://github.com/vgvassilev/clad or https://clad.readthedocs.io/en/latest/index.html. Maybe I missed it. Is there also a way to remove the output of `The code is:`?

**Comment by vgvassilev** [comment]
One way to work this around is to add `-Xclang -plugin-arg-clad -Xclang -fgenerate-source-file`. This would produce a file `Derivatives.cpp` which should contain all information. Note that Clad appends to that file, so you will have to clean it by hand every run.

EDIT: > Is there also a way to remove the output of The code is:?

Yes, calling `getCode()`.

**Reply by MaxSagebaum** [reply]
Thanks for the quick answer. That solvers my problem. It also is part of the answer to my earlier question: _If clad could also be used as a standalone tool?_

With this I no longer need to generate an executable, calling the compiler is enough.

Are these flags documented somewhere?

**Reply by vgvassilev** [reply]
In the testsuite :) But it'd be great if you open issues for the non-documented parts and possibly tell us where do you expect to see them appearing...

**Reply by MaxSagebaum** [reply]
Thanks, I created the new issure https://github.com/vgvassilev/clad/issues/1156.

**Reply by MaxSagebaum** [reply]
I can not mark this discussion as answered. Can you do it?
